app/src/org/commcare/network/HttpRequestGenerator.java

```diff
@@ -168,6 +169,12 @@ public HttpResponse makeCaseFetchRequest(String baseUri, boolean includeStateFla
         //Add items count to fetch request
         serverUri = serverUri.buildUpon().appendQueryParameter("items", "true").build();
 
+        if (CommCareApplication._().shouldInvalidateCacheOnRestore()) {
+            serverUri = serverUri.buildUpon().appendQueryParameter("overwrite_cache", "true").build();

```

Comment:
Yeah, I think the why is very useful in this context. For instance, I have no idea how `serverUri.buildUpon().appendQueryParameter("items", "true").build();` is used by HQ and since it isn't in our code base, I can't just go look it up.